File: PRACTICA 3/JAVA/Cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int cantidad;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.cantidad = 0;
    }

    public void insert(long e) {
        if (isFull()) {
            System.out.println("Error: Cola llena");
            return;
        }
        fin = (fin + 1) % n;
        arreglo[fin] = e;
        cantidad++;
    }

    public long remove() {
        if (isEmpty()) {
            System.out.println("Error: Cola vacía");
            return -1;
        }
        long valor = arreglo[inicio];
        inicio = (inicio + 1) % n;
        cantidad--;
        return valor;
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Error: Cola vacía");
            return -1;
        }
        return arreglo[inicio];
    }

    public boolean isEmpty() {
        return cantidad == 0;
    }

    public boolean isFull() {
        return cantidad == n;
    }

    public int size() {
        return cantidad;
    }
}

File: PRACTICA 3/PYTHON/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.cantidad = 0

    def insert(self, e):
        if self.isFull():
            print("Error: Cola llena")
            return
        self.fin = (self.fin + 1) % self.n
        self.arreglo[self.fin] = e
        self.cantidad += 1

    def remove(self):
        if self.isEmpty():
            print("Error: Cola vacía")
            return None
        valor = self.arreglo[self.inicio]
        self.inicio = (self.inicio + 1) % self.n
        self.cantidad -= 1
        return valor

    def peek(self):
        if self.isEmpty():
            print("Error: Cola vacía")
            return None
        return self.arreglo[self.inicio]

    def isEmpty(self):
        return self.cantidad == 0

    def isFull(self):
        return self.cantidad == self.n

    def size(self):
        return self.cantidad


if __name__ == "__main__":
    cola = Cola(5)
    cola.insert(10)
    cola.insert(20)
    cola.insert(30)
    print("Frente:", cola.peek())
    print("Elemento eliminado:", cola.remove())
    print("Nuevo frente:", cola.peek())
    print("¿Está vacía?", cola.isEmpty())
    print("Tamaño actual:", cola.size())
